apollo-router/src/query_planner/dual_query_planner.rs

```diff
@@ -879,6 +900,48 @@ mod ast_comparison_tests {
         assert!(super::same_ast_document(&ast_x, &ast_y).is_ok());
     }
 
+    #[test]
+    fn test_query_variable_decl_object_value_coercion_non_empty_case() {
+        // Note: Rust QP expands an object default values by filling in its default field values.
+        let op_x = r#"query($qv1: T! = {field1: true}) { x(arg1: $qv1) }"#;
+        let op_y =
+            r#"query($qv1: T! = { field1: true, field2: "default_value" }) { x(arg1: $qv1) }"#;
+        let ast_x = ast::Document::parse(op_x, "op_x").unwrap();
+        let ast_y = ast::Document::parse(op_y, "op_y").unwrap();
+        assert!(super::same_ast_document(&ast_x, &ast_y).is_ok());
+    }
+
+    #[test]
+    fn test_query_variable_decl_list_value_coercion_empty_case() {
+        // Note: Rust QP expands empty list default values by filling in its default items.
+        let op_x = r#"query($qv1: [Int!]! = []) { x(arg1: $qv1) }"#;
+        let op_y = r#"query($qv1: [Int!]! = [1, 42]) { x(arg1: $qv1) }"#;
```

Comment:
This seems strange? these are not equivalent queries.

Reply:
`y` is expected to have additional items over `x` by filling in the (assumed-to-be) default value.

Reply:
I'm confused about that. The default value is declared to be `[]` in one and `[1, 42]` in the other. This isn't a list that would pull in default values--it's a declaration of a default value.

Reply:
For objects, default property values come from the input object definition, but that doesn't exist for lists

Reply:
Ah, you are right. We don't have to loosen lists.